### PR TITLE
Make MonsterModel ready to replace DgMonster fully

### PR DESCRIPTION
**Pull request by RheingoldRiver**
so far i just made an EvolutionModel and wrote some test cases, also it won't run because i commented the sqlite path to run the test cases

### CODE REVIEW

**Comment by RheingoldRiver** [comment]
Before this change to MonsterIndex, `^id uuvo Typhon` was broken, it was returning reinc typhon instead of typhon ana for example

**Comment by RheingoldRiver** [comment]
```python
    def get_monsters_by_awakenings(self, awoken_skill_id: int):
        # TODO: Make this not make monsters via query
        return self.database.query_many(
            self.database.select_builder(
                tables=OrderedDict({
                    DgMonster.TABLE: DgMonster.FIELDS,
                    DgAwakening.TABLE: None,
                }),
                where='{}.awoken_skill_id=?;'.format(DgAwakening.TABLE),
                key=('monster_id',),
                distinct=True
            ),
            (awoken_skill_id,),
            DgMonster, graph=self.graph)
```
this method is still in DbContext despite not being used at all....should I kill it? we currently cannot do this from the graph but it's maybe useful? idk? should I add support for it somehow? update it to return monster models & save it? something else?

**Comment by RheingoldRiver** [comment]
> 
> 
> Before this change to MonsterIndex, `^id uuvo Typhon` was broken, it was returning reinc typhon instead of typhon ana for example

Actually see other commit messages, it was broken twice over - both the MonsterIndex stuff was wrong, and *also* the timestamp stuff wasn't being respected, gg rip

**Comment by RheingoldRiver** [comment]
Ok PR is now potentially done pending aradia trying to break it

File: dadguide/models/evolution_model.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    def __init__(self, **kwargs):
+        self.evolution_type = kwargs['evolution_type']
+        self.from_id = kwargs['from_id']
+        self.to_id = kwargs['to_id']
+        self.mat_1_id = kwargs['mat_1_id']
+        self.mat_2_id = kwargs['mat_2_id']
+        self.mat_3_id = kwargs['mat_3_id']
+        self.mat_4_id = kwargs['mat_4_id']
+        self.mat_5_id = kwargs['mat_5_id']
+        mats = [self.mat_1_id, self.mat_2_id, self.mat_3_id, self.mat_4_id, self.mat_5_id]
```
Comment (TsubakiBotPad) [inline dadguide/models/evolution_model.py:14]: maybe just `self.mats = filter(None, [self.mat_1_id, self.mat_2_id, self.mat_3_id, self.mat_4_id, self.mat_5_id])`

Comment (RheingoldRiver) [reply dadguide/models/evolution_model.py:14]: omg list(filter()) u suck

**Pull request merged**